editoast/src/generated_data/error/mod.rs

```diff
@@ -33,16 +85,64 @@ impl GeneratedData for ErrorLayer {
         let graph = Graph::load(infra_cache);
 
         // Generate the errors
-        track_sections::insert_errors(conn, infra, infra_cache, &graph)?;
-        signals::insert_errors(conn, infra, infra_cache)?;
-        speed_sections::insert_errors(conn, infra, infra_cache)?;
+
+        let infra_errors = generate_errors(
+            ObjectType::TrackSection,
+            infra_cache,
+            &graph,
+            &track_sections::TRACK_SECTIONS_ERRORS,
+        );
+        track_sections::insert_errors(infra_errors, conn, infra)?;
+        let infra_errors = generate_errors(
+            ObjectType::Signal,
+            infra_cache,
+            &graph,
+            &signals::SIGNALS_ERRORS,
+        );
+        signals::insert_errors(infra_errors, conn, infra)?;
+        let infra_errors = generate_errors(
+            ObjectType::SpeedSection,
+            infra_cache,
+            &graph,
+            &speed_sections::SPEED_SECTION_ERRORS,
+        );
+        speed_sections::insert_errors(infra_errors, conn, infra)?;
+
         track_section_links::insert_errors(conn, infra, infra_cache)?;
-        switch_types::insert_errors(conn, infra, infra_cache)?;
+
+        let infra_errors = generate_errors(
+            ObjectType::SwitchType,
+            infra_cache,
+            &graph,
+            &switch_types::SWITCH_TYPES_ERRORS,
+        );
+        switch_types::insert_errors(infra_errors, conn, infra)?;
+
         switches::insert_errors(conn, infra, infra_cache)?;
-        detectors::insert_errors(conn, infra, infra_cache)?;
-        buffer_stops::insert_errors(conn, infra, infra_cache)?;
+
+        let infra_errors = generate_errors(
+            ObjectType::Detector,
+            infra_cache,
+            &graph,
+            &detectors::DETECTOR_ERRORS,
+        );
+        detectors::insert_errors(infra_errors, conn, infra)?;
+        let infra_errors = generate_errors(
+            ObjectType::BufferStop,
+            infra_cache,
+            &graph,
+            &buffer_stops::BUFFER_STOP_ERRORS,
+        );
+        buffer_stops::insert_errors(infra_errors, conn, infra)?;
+
         routes::insert_errors(conn, infra, infra_cache, &graph)?;
```

Comment:
Can you separate objects like `routes`, `switches`, `track_section_links` that uses the old way to generate errors. You can also add a comment to explain why it's done differently for now.